pkg/haproxy/instance_test.go

```diff
@@ -4922,18 +4951,35 @@ func TestModSecurity(t *testing.T) {
 backend spoe-modsecurity
     mode tcp` + test.modsecExp
 		}
-		c.checkConfig(`
+		// unique-id-format must be set when using Coraza
+		if test.modsecUseCoraza {
+			c.checkConfig(`
 <<global>>
 <<defaults>>
+    unique-id-format        %[uuid()]
 backend d1_app_8080
     mode http` + test.backendExp + `
     server s1 172.17.0.11:8080 weight 100
 <<backends-default>>
 <<frontends-default>>
 <<support>>` + modsec)
+		} else {
+			c.checkConfig(`
+<<global>>
+<<defaults>>
+backend d1_app_8080
+    mode http` + test.backendExp + `
+    server s1 172.17.0.11:8080 weight 100
+<<backends-default>>
+<<frontends-default>>
+<<support>>` + modsec)
+		}
 		if test.modsecAgentExp != "" {
 			c.containsText("spoe-modsecurity.conf", c.readConfig(c.tempdir+"/spoe-modsecurity.conf"), test.modsecAgentExp)
 		}
+		if test.modsecOtherExp != "" {
+			c.containsText("spoe-modsecurity.conf", c.readConfig(c.tempdir+"/spoe-modsecurity.conf"), test.modsecOtherExp)
+		}
```

Comment:
Do we need both `modsecAgentExp` and `modsecOtherExp`?

Reply:
Yes `modsecOtherExp` is just another contiguous substring to search for inside the templated file. Without it, the tests that check `var-prefix` would need to basically include a full copy of `spoe-modsecurity.conf` since the `var-prefix` and the `args` are so far apart in the file.

Reply:
Ah sure it's a substring. Nevermind. Just missing a local e2e and we're done.